### PR TITLE
Updating account details

### PR DESCRIPTION
**Pull request by vytis**
Added listening for user changes so that we could get the team name on initial launch. Team name changes are observed as well and updated accordingly. 
Currently the team picture is not downloaded here and is nil in account.

Needs release with:
- [x] https://github.com/wireapp/wire-ios-data-model/pull/354

### CODE REVIEW

File: Source/SessionManager/SessionManager.swift [hunk 1]
```diff
@@ -327,6 +335,43 @@ public typealias LaunchOptions = [UIApplicationLaunchOptionsKey : Any]
...
+        guard let session = userSession else { return }
+        let selfUser = ZMUser.selfUser(in: session.syncManagedObjectContext)
+        if let account = accountManager.accounts.first(where: { $0.userIdentifier == selfUser.remoteIdentifier }) {
+            if let name = team?.name {
+                account.teamName = name
+            }
+            if let userName = userName {
+                account.userName = userName
+            }
+            accountManager.add(account)
```
Comment (typfel) [inline Source/SessionManager/SessionManager.swift]: Adding the `account` will update it if it already exists?

File: Source/SessionManager/SessionManager.swift [hunk 2]
```diff
@@ -273,9 +275,12 @@ public typealias LaunchOptions = [UIApplicationLaunchOptionsKey : Any]
         guard let session = authenticatedSessionFactory.session(for: account, storeProvider: provider) else {
             preconditionFailure("Unable to create session for \(account)")
         }
+        let selfUser = ZMUser.selfUser(inUserSession: session)
+        teamObserver = TeamChangeInfo.add(observer: self, for: nil)
+        selfObserver = UserChangeInfo.add(observer: self, forBareUser: selfUser!)
```
Comment (typfel) [inline Source/SessionManager/SessionManager.swift]: Shouldn't we release the observers when logging out or switching accounts?

File: Source/SessionManager/SessionManager.swift (hunk 1 above)
Comment (vytis) [reply Source/SessionManager/SessionManager.swift]: Yes, it is basically `createOrUpdate` under the hood, only named slightly confusing. Also has tests for update behaviour in DataModel.

File: Source/SessionManager/SessionManager.swift (hunk 2 above)
Comment (vytis) [reply Source/SessionManager/SessionManager.swift]: Good point

**Comment by vytis** [comment]
@typfel updated

**Comment by mikeger** [comment]
@vytis we are not setting the `imageData` on the account yet? We could use the user image for non-team users.

**Comment by typfel** [comment]
What about if the account already exists? I guess the name wouldn't get set until it's updated?

**Comment by vytis** [comment]
@typfel I think it should update with the supplied name then - https://github.com/wireapp/wire-ios-data-model/blob/develop/Tests/Source/Model/AccountStoreTests.swift#L157

**Comment by typfel** [comment]
@vytis I phrased it weirdly. I mean when the scenario when you upgrade and you already have a "synced" account. Then you wouldn't get any change notifications.

**Comment by vytis** [comment]
@typfel update the code. The tests are lacking for now, but that needs a bit of refactoring to be able to inject dependencies.

**Comment by typfel** [comment]
@mikeger tests were still failing

**Comment by mikeger** [comment]
Sorry, I implied from comment that the tests are missing and not failing. From the log I see that the test task hanged. @vytis @typfel  should we revert?

**Pull request merged**